.github/workflows/lint.yml

```diff
@@ -0,0 +1,30 @@
+name: Lint
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run Black
+        uses: psf/black@stable
+        with:
+          options: "--check --verbose"
+          src: "./src"
+          version: 24.8.0
+      - name: Setup Python 

```

Comment:
Why does setup python happen after black runs? Logically, I'd expect setup (and installing deps) to happen first, then the linter and formatter get run.

Reply:
Black has a first party ci/cd workflow which handles the installation and setup for us. mypy and pylint dont and thus we need to setup python and install them.

Reply:
Gotcha. Could you please add a comment before the black line explaining that? Thanks!